benches/benches/block_target_gas_set/contract.rs

```diff
@@ -1,20 +1,741 @@
+        run_with_service(
+            &id,
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+
+    // log
+    {
+        run(
+            "contract/log",
+            group,
+            vec![op::log(0x10, 0x11, 0x12, 0x13), op::jmpb(RegId::ZERO, 0)],
+            vec![],
+        );
+    }
+
+    // logd
+    {
+        for i in arb_dependent_cost_values() {
+            let mut instructions = setup_instructions();
+            instructions.extend(vec![
+                op::movi(0x13, i),
+                op::logd(0x10, 0x11, RegId::ZERO, 0x13),
+                op::jmpb(RegId::ZERO, 0),
+            ]);
+            let id = format!("contract/logd {:?}", i);
+            run_with_service(
+                &id,
+                group,
+                instructions,
+                script_data.clone(),
+                &service,
+                contract_id,
+                &rt,
+                &mut rng,
+            );
+        }
+    }
+
+    // mint
+    {
+        let contract = vec![op::mint(RegId::ONE, RegId::ZERO), op::ret(RegId::ZERO)];
+        let instructions = call_contract_repeat();
+        replace_contract_in_service(&mut service, &contract_id, contract);
+        run_with_service(
+            "contract/mint",
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+
+    // ret contract
+    {
+        let contract = vec![op::ret(RegId::ONE), op::ret(RegId::ZERO)];
+        let instructions = call_contract_repeat();
+        replace_contract_in_service(&mut service, &contract_id, contract);
+        run_with_service(
+            "contract/ret contract",
+            group,
+            instructions,
+            script_data.clone(),
+            &service,
+            contract_id,
+            &rt,
+            &mut rng,
+        );
+    }
+
+    // retd contract
+    {
+        for i in arb_dependent_cost_values() {
+            let contract = vec![op::movi(0x14, i), op::retd(RegId::ONE, 0x14)];
+            let instructions = call_contract_repeat();
+            replace_contract_in_service(&mut service, &contract_id, contract);
+            let id = format!("contract/retd contract {:?}", i);
+            run_with_service(
+                &id,
+                group,
+                instructions,
+                script_data.clone(),
+                &service,
+                contract_id,
+                &rt,
+                &mut rng,
+            );
+        }
+    }
+
+    //     run_group_ref(
+    //         &mut c.benchmark_group("rvrt_contract"),
+    //         "rvrt_contract",
+    //         VmBench::contract(rng, op::ret(RegId::ONE)).unwrap(),
+    //     );
+
+    // TODO: Is `rvrt` even possible to bench?
```

Comment:
Perhaps we can create a separate ticket to investigate benchmarks for `rvrt`. Depending on the results of that investigation, we can continue from there.

Reply:
Yeah. By its nature it is going to revert everything you've done. It's not important to bench in that case, since there is no way to spam it. We can just cost it something reasonable.